internal/experiments/experiments_ai_data_leakage.go

```diff
@@ -0,0 +1,226 @@
+package experiments
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/operantai/secops-chaos/internal/categories"
+	"github.com/operantai/secops-chaos/internal/k8s"
+	"github.com/operantai/secops-chaos/internal/verifier"
+	"gopkg.in/yaml.v3"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type LLMDataLeakageExperiment struct {
+	Metadata   ExperimentMetadata `yaml:"metadata"`
+	Parameters LLMDataLeakage     `yaml:"parameters"`
+}
+
+type LLMDataLeakage struct {
+	Apis []ExecuteAIAPI `yaml:"apis"`
+}
+
+type AIAPIPayload struct {
+	Model                string   `json:"model" yaml:"model"`
+	AIApi                string   `json:"ai_api" yaml:"ai_api"`
+	SystemPrompt         string   `json:"system_prompt" yaml:"system_prompt"`
+	Prompt               string   `json:"prompt" yaml:"prompt"`
+	VerifyPromptChecks   []string `json:"verify_prompt_checks" yaml:"verify_prompt_checks"`
+	VerifyResponseChecks []string `json:"verify_response_checks" yaml:"verify_response_checks"`
+}
+
+type AIVerifierResult struct {
+	Check      string  `json:"check"`
+	Detected   bool    `json:"detected"`
+	EntityType string  `json:"entityType"`
+	Score      float64 `json:"score"`
+}
+
+type AIAPIResponse struct {
+	VerifiedPromptChecks   []AIVerifierResult `json:"verified_prompt_checks" yaml:"verified_prompt_checks"`
+	VerifiedResponseChecks []AIVerifierResult `json:"verified_response_checks" yaml:"verified_response_checks"`
+}
+
+type ExecuteAIAPI struct {
+	Description      string        `yaml:"description"`
+	Payload          AIAPIPayload  `yaml:"payload"`
+	ExpectedResponse AIAPIResponse `yaml:"expected_response"`
+}
+
+type ExecuteAIAPIResult struct {
+	ExperimentName string        `json:"experiment_name"`
+	Description    string        `json:"description"`
+	Timestamp      time.Time     `json:"timestamp"`
+	Status         int           `json:"status"`
+	Response       AIAPIResponse `json:"response"`
+}
+
+func (p *LLMDataLeakageExperiment) Type() string {
+	return "llm_data_leakage"
+}
+
+func (p *LLMDataLeakageExperiment) Description() string {
+	return "This experiment checks whether the LLM AI Model is leaking any sensitive data such as PII data or secrets and keys in its response"
+}
+func (p *LLMDataLeakageExperiment) Technique() string {
+	return categories.MITREATLAS.Exfiltration.LLMDataLeakage.Technique
+}
+func (p *LLMDataLeakageExperiment) Tactic() string {
+	return categories.MITREATLAS.Exfiltration.LLMDataLeakage.Tactic
+}
+func (p *LLMDataLeakageExperiment) Framework() string {
+	return string(categories.MitreAtlas)
+}
+
+func (p *LLMDataLeakageExperiment) Run(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) error {
```

Comment:
Should we add some guards here to ensure that the secops-chaos-ai component is deployed?